### ISSUE TITLE
Mac version problems from v1.1.0

### ISSUE THREAD

**Issue by zotil** [actionquake/distrib#260]
Hi guys, I've been playing with one of the first versions and I didn't update from long. Today I installed the last version (1.2.6) but I could'nt open the game. Here is the relevant info from the crash report:

```
Termination Reason:    Namespace DYLD, Code 1 Library missing
Library not loaded: '@loader_path/discord_game_sdk.dylib'
Referenced from: '/Applications/AQ.app/Contents/MacOS/q2pro'
Reason: tried: '' (no such file), '' (no such file), '' (no such file)
(terminated at launch; ignore backtrace)
```

Then I tried to downgrade, but I got the same problem until version 1.2.0

Continuing the downgrade, I got a different problem running the 1.1.x versions:

```
------- FS_Init -------
Current search path:
./AQ.app/Contents/MacOS/action
./AQ.app/Contents/MacOS/action
-----------------------
------- S_Init -------
Using SDL audio driver: coreaudio
sound sampling rate: 22050
----------------------
------- R_Init -------
Detected OpenGL 2.1
Loaded extension GL_EXT_texture_filter_anisotropic
Loaded extension GL_ARB_fragment_program
Using legacy rendering backend.
********************
FATAL: Couldn't load pics/colormap.pcx: No such file or directory
********************
```

The last working version for me is 1.0.18 which I have installed now and I can open without problems.

macOS Monterrey 12.5.1 (Intel)
Discord installed

**Comment by zotil** [comment]
I installed the game from steam and didn't have problems.

![imagen](https://user-images.githubusercontent.com/18425214/189707206-f2ea5a48-1aa1-438f-9644-3f8b7836a628.png)

I could not be able to run the game installing from dmg.

**Comment by zotil** [comment]
I copied the discord_game_sdk.dylib from the steam version the latest dmg release version. It doesn't crash but it doesn't open.

Trying to run q2pro inside of /Applications/AQ.app/Contents/MacOS path, gives me the same colormap.pcx error. I tried with +set game action but no luck. I am missing something?

**Comment by zotil** [comment]
Ok I tried replacing the paks without luck, finally I replaced the q2pro binary from the steam version the latest dmg release version.

Now I can run q2pro from terminal but can't open AQ.app. I get the following log in the launchd.log

```
2022-09-12 16:01:42.681881 (gui/501/application.com.aq2world.16413079.16413184 [23117]) <Notice>: exited due to exit(1)
2022-09-12 16:01:42.681899 (gui/501/application.com.aq2world.16413079.16413184 [23117]) <Notice>: service state: exited
2022-09-12 16:01:42.681905 (gui/501/application.com.aq2world.16413079.16413184 [23117]) <Notice>: internal event: EXITED, code = 0
2022-09-12 16:01:42.681910 (gui/501/application.com.aq2world.16413079.16413184 [23117]) <Notice>: job state = exited
2022-09-12 16:01:42.681935 (gui/501 [100003]) <Notice>: service inactive: application.com.aq2world.16413079.16413184
2022-09-12 16:01:42.681941 (gui/501/application.com.aq2world.16413079.16413184 [23117]) <Notice>: service state: not running
2022-09-12 16:01:42.683203 (gui/501/application.com.aq2world.16413079.16413184) <Notice>: removing job: caller = runningboardd
2022-09-12 16:01:42.683207 (gui/501 [100003]) <Notice>: removing service: application.com.aq2world.16413079.16413184
2022-09-12 16:01:42.683277 (gui/501/application.com.aq2world.16413079.16413184) <Notice>: internal event: PETRIFIED, code = 0
2022-09-12 16:01:42.683280 (gui/501/application.com.aq2world.16413079.16413184) <Notice>: job state = removed
```

Resuming:
- discord_game_sdk.dylib was missing. Copying from steam version does the trick.
- q2pro binary gives pic/colormap.pcx error. Replacing for steam version work fine.
- AQ.app doesn't open after the above changes, but works calling ./q2pro

**Comment by darkshade9** [comment]
Thanks for this report @zotil  -- this is a problem that only affects the Standalone Mac version, as noted.  I'm toying with the idea of deprecating the standalone versions as it's a significant amount of work to keep it operational and the Steam version works just fine.  I'll check and see if this is an easy fix though, in the meantime, I highly suggest all Mac players use Steam, It Just Works(tm)

**Comment by Raptor007** [comment]
> * discord_game_sdk.dylib was missing. Copying from steam version does the trick.
> 
> * q2pro binary gives pic/colormap.pcx error. Replacing for steam version work fine.
> 
> * AQ.app doesn't open after the above changes, but works calling ./q2pro

Maybe the AQ.app is code-signed?  If so, changing files within it would cause the signature to no longer be valid.  You could try creating a self-signed root certificate in your Keychain, and then `codesign -s "Name of Your Cert" "AQ.app"` to re-sign the app.

**Comment by zotil** [comment]
> Maybe the AQ.app is code-signed? If so, changing files within it would cause the signature to no longer be valid. You could try creating a self-signed root certificate in your Keychain, and then codesign -s "Name of Your Cert" "AQ.app" to re-sign the app.

I forgot that part, as I don't work often with mac apps, I just started to move files to make the game open without signing the app.

Today I tested version 1.2.6 which comes on a zip file instead of standalone app.

As for me, I am happy with this version, as it's more versatile to config and update.As for me, I am happy with this version, as it's more versatile to config and update.

_(I leave this information in case is useful.)_
I tried executing launch.sh but I had to give permission to aqtion executable first (unverified developer), so I did. I did run launch.sh again but it failed because I was not running steam. I don't use steam so I ran q2pro (must give permission again) and it failed because discord_game_sdk.dylib also needs permission.
After giving permissions to this 3 files (aqtion, q2pro, discord_game_sdk.dylib) it run ok.
